### ISSUE TITLE
Effect of long wav files?

### ISSUE THREAD

**Issue by reiinakano** [CSTR-Edinburgh/merlin#174]
Would there be a benefit to split long (30 seconds-ish) wav files to smaller ones? What is the optimum length for a wav file that is fed to Merlin?

**Comment by ronanki** [comment]
As long as the alignments are fine, there won't be any problem in training a DNN. However, RNNs may not give optimal performance -- switching to keras training (set switch_to_keras as True) may help.

**Comment by reiinakano** [comment]
Sorry where is the switch_to_keras option? I've looked in duration_my_voice.conf and acoustic_my_voice.conf but I don't see it there.

**Comment by ronanki** [comment]
Sorry about this -- but you need to pull the latest code, to enable switch_to_keras option. 

The default conf files are just for an example -- they are not recommended settings. One should always try to optimise the hyper parameters as per their data for training either DNNs or RNNs. 

RNNs may offer either partial or substantial improvement depending on the data. To switch from DNNs to RNNs, you need to set couple of layers from TANH to LSTM in hidden_layer_type and set sequential_training to True. 

You also need to change the batch_size as well from 256 to 25 or some number less than 50 (i.e., number of sentences to process in a batch) . Merlin by default uses batch size one for RNNs and takes huge amount of time. However, batch training for RNNs can be enabled now using Keras option (i.e., when switch_to_keras is set to True).

There are no resources as of now -- we'll be providing soon, but you can change any parameter you wish. For alignments, use either audacity or wavesurfer to open the audio file and load the label file (label file format differs).

**Comment by reiinakano** [comment]
So if I set switch to keras as True, I can use RNNs with my long wav files?

**Comment by ronanki** [comment]
If you set switch_to_keras as True -- training becomes much faster (may be 20 times). For this, you need to install keras and sklearn libraries. 

However, you can also use RNNs even without switching, but takes a day or two to train depending on the amount of data.

**Comment by m-toman** [comment]
Really that much faster? Is the Tensorflow backend so much faster than Theano or is there another reason?

**Comment by reiinakano** [comment]
I've updated merlin and still can't see `switch_to_keras`. Should I just insert it in acoustic and duration configuration files? By the way, should I keep architecture + hyperparams for duration and acoustic model the same?

**Comment by ronanki** [comment]
@m-toman Merlin by default uses batch size one for training RNNs, as the utterances vary in length. Now, batch processing was implemented when using Keras Models, with few data processing techniques -- that means, you can process 50 sentences (for example) in one batch. This enables faster training and better optimisation -- and with Keras you can use either theano or tensorflow as backend. Both seems to perform equally good in terms of computation speed and performance (with tensorflow slightly better).  

@reiinakano Just insert in acoustic and duration config files under architecture section. Make sure, you are using [latest code](https://github.com/CSTR-Edinburgh/merlin/blob/master/src/run_merlin.py#L81).

**Comment by reiinakano** [comment]
Is it fine to have different architectures for acoustic and duration DNN?  For example, duration DNN switch_to_keras=False while acoustic DNN switch_to_keras=True? Or is it better to have the same hyperparameters+architecture for both?
